src/main/scala/scorex/core/api/http/ApiResponse.scala

```diff
@@ -2,35 +2,43 @@ package scorex.core.api.http
 
 import akka.http.scaladsl.model.{ContentTypes, HttpEntity, StatusCode, StatusCodes}
 import akka.http.scaladsl.server.{Directives, Route}
-import io.circe.Json
+import io.circe.{Encoder, Json}
 import io.circe.syntax._
 
 import scala.concurrent.Future
 import scala.language.implicitConversions
 
+/** Bunch of methods to wrap json result to route with the given status code, completing it.
+  * When receiving `Null` json, then complete route with `404 not found` status code.
+  */
 class ApiResponse(statusCode: StatusCode) {
-  def apply(result: Future[Json]): Route = Directives.onSuccess(result)(complete)
+
   def apply(result: Json): Route = withJson(result)
+  def apply[R: Encoder](result: R): Route = withJson(result)
+  def apply[R: Encoder](result: Future[R]): Route = Directives.onSuccess(result)(withJson)
+  def apply(result: Future[Json]): Route = Directives.onSuccess(result)(withJson)
 
   def defaultRoute: Route = withString(defaultMessage)
   def defaultMessage: String = statusCode.reason
 
   def withString(s: String): Route = complete(s.asJson)
-  def withJson(result: Json): Route = complete(result)
 
-  def complete(result: Json): Route = {
-    Directives.complete(statusCode.intValue() -> HttpEntity(ContentTypes.`application/json`, result.spaces2))
+  def withJson[R](result: R)(implicit encoder: Encoder[R]): Route = complete(encoder(result))
+
+  def complete(result: Json): Route = result match {
+    case json if json.isNull => ApiError.NotExists
+    case _ => Directives.complete(statusCode.intValue() -> HttpEntity(ContentTypes.`application/json`, result.spaces2))
```

Comment:
minor comment, the match expression could be replaced by an if expression
```
if(result.isNull) { 
  ApiError.NotExists
} else {
  Directives.complete(statusCode.intValue() -> HttpEntity(ContentTypes.`application/json`, result.spaces2))
}
```

Reply:
I see, I would favor the if expression based on this discussion (mainly efficiency) https://stackoverflow.com/questions/9266822/pattern-matching-vs-if-else
I personally also prefer it on style, but my taste is not a good reason for changing this.

@kushti @ceilican 
is there any reason why we see/prefer pattern matching over if expressions as @mike-aksarin  noticed in the code?